Authorization/Authorization/Presentation/SocialSign/SocialSignViewModel.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  SocialSignViewModel.swift
+//  Authorization
+//
+//  Created by Eugene Yatsenko on 11.10.2023.
+//
+
+import SwiftUI
+import Core
+import AuthenticationServices
+import FacebookLogin
+import GoogleSignIn
+import MSAL
+import Swinject
+
+enum SocialResult {
+    case apple(AppleCredentials)
+    case facebook(LoginManagerLoginResult)
+    case google(GIDSignInResult)
+    case microsoft(MSALAccount, String)
+
+    var backend: String {
+        switch self {
+        case .apple:
+            return "apple-id"
+        case .facebook:
+            return "facebook"
+        case .google:
+            return "google-oauth2"
+        case .microsoft:
+            return "azuread-oauth2"
+        }
+    }
+}
+
+final public class SocialSignViewModel: ObservableObject {
+
+    // MARK: - Properties
+
+    private var completion: ((Result<SocialResult, Error>) -> Void)
+    private let config: ConfigProtocol
+
+    init(
+        config: ConfigProtocol,
+        completion: @escaping (Result<SocialResult, Error>) -> Void
+    ) {
+        self.config = config
+        self.completion = completion
+    }
+
+    private let appleSingInProvider: AppleSingInProvider  = .init()
+    private let googleSingInProvider: GoogleSingInProvider = .init()
+    private let facebookSingInProvider: FacebookSingInProvider = .init()
+    private let microsoftSingInProvider: MicrosoftSingInProvider = .init()
```

Comment:
We should not initialize dependencies manually, this is a violation of the SOLID principles.
We have to provide dependencies to the initializer block via DI.

You probably won't be able to write tests for this class if you create dependencies manually.

Reply:
After discussion decided to leave it as is for simplicity